### ISSUE TITLE
API key expired Refresh too fast.

### ISSUE THREAD

**Issue by snkiz** [Elinvention/gnome-shell-extension-nasa-apod#14]
Hello I'm using Ubuntu 17.10 and GNOME 3.26. After the latest update I'm getting this error;

NASA APOD extension: https://api.nasa.gov/planetary/apod?api_key=XKSoS8Bv05ij8JH8UWa7eqMavXgGfFStcc6Pu3KH&date=Total-solar-eclipse
NASA APOD extension error: Over rate limit (error 429): Get your API key at https://api.nasa.gov/ to have 1000 requests per hour just for you. Will retry in 5 minutes.
NASA APOD extension: Next check in 300 seconds @ local time 14:05
NASA APOD extension: Refresh done.

I don't recall ever getting my own key, I'm guessing this one is burned now. do I need to get a key to keep it working now? 300 seconds is insane, it not a weather app. 43200 is 12 hours, better yet just have it refresh once a day. It is the image of the day after all, not the image of 5 min.

**Comment by Elinvention** [comment]
One key can be used at most 1000 times per hour. The extension comes bundled with my key, but a lot of people use the extension, so more than 1000 requests are made per hour with my key. When that happens the server returns error 429 and the new image can't be downloaded. If you try again in 5 minutes, usually the counter is below 1000, allowing you to download the picture of the day.

What the notification says is that you can register your own key IF you want to avoid this problem, but you are not required. Just know that you'll have to wait if more than 1000 requests were made the last hour. Previously the extension didn't check for this error and notified about a generic network error, which, I think, was misleading.

**Comment by snkiz** [comment]
Makes sense, and I appreciate the more informative error message. I'm not the one trying again in 5 min however, the extension is. I don't see anyway to change that without editing the code. Sure I can do that, but a IMO 300 second timer seems excessive as default. The other unintended consequence, I've found (and only now figured out.) is that if I manually set an older image because say I don't like today's image it gets changed back pretty quickly. It is mildly infuriating.
